validator/client/runner.go

```diff
@@ -175,3 +185,34 @@ func handleAssignmentError(err error, slot uint64) {
 		log.WithField("error", err).Error("Failed to update assignments")
 	}
 }
+
+func handleAccountsChanged(ctx context.Context, v Validator, accountsChangedChan chan struct{}) {
+	validatingPubKeysChan := make(chan [][48]byte, 1)
+	var sub event.Subscription
+	switch km := v.GetKeymanager().(type) {
```

Comment:
Should this function instead take in a keymanager iface as an argument instead of the `validator` ? Also, we can consider adding SubscribeAccountChanges as a method for the interface, and for remote wallets, simply return error cannot subscribe to account changes. This will make it easier for you to avoid having to do this type assertion. Also, it will make this function really easy to test as all you will need is a mock for the small keymanager interface and can be unit tested. I don't see why tests are blocked for this function if we change these items

Reply:
We can't throw an error because it will make remote wallets unusable - you would get an error every time. We could make it a no-op, though. That would indeed simplify everything.